giraffe/src/utils/orderBy.ts

```diff
@@ -0,0 +1,188 @@
+import {get} from './get'
+import {isLength} from './isLength'
+import {isSymbol} from './isSymbol'
+import {isString} from './isString'
+
+const baseSortBy = (array, comparer) => {
+  let {length} = array
+
+  array.sort(comparer)
+  while (length--) {
+    array[length] = array[length].value
+  }
+  return array
+}
+
+const identity = value => value
+
+const compareAscending = (value, other) => {
+  if (value !== other) {
+    const valIsDefined = value !== undefined
+    const valIsNull = value === null
+    const valIsReflexive = value === value
+    const valIsSymbol = isSymbol(value)
+
+    const othIsDefined = other !== undefined
+    const othIsNull = other === null
+    const othIsReflexive = other === other
+    const othIsSymbol = isSymbol(other)
+
+    const val =
+      typeof value === 'string' ? value.localeCompare(other) : value - other
+
+    if (
+      (!othIsNull && !othIsSymbol && !valIsSymbol && val > 0) ||
+      (valIsSymbol &&
+        othIsDefined &&
+        othIsReflexive &&
+        !othIsNull &&
+        !othIsSymbol) ||
+      (valIsNull && othIsDefined && othIsReflexive) ||
+      (!valIsDefined && othIsReflexive) ||
+      !valIsReflexive
+    ) {
+      return 1
+    }
+    if (
+      (!valIsNull && !valIsSymbol && !othIsSymbol && val < 0) ||
+      (othIsSymbol &&
+        valIsDefined &&
+        valIsReflexive &&
+        !valIsNull &&
+        !valIsSymbol) ||
+      (othIsNull && valIsDefined && valIsReflexive) ||
+      (!othIsDefined && valIsReflexive) ||
+      !othIsReflexive
+    ) {
+      return -1
+    }
+  }
+  return 0
+}
+
+const compareMultiple = (object, other, orders) => {
+  let index = -1
+  const objCriteria = object.criteria
+  const othCriteria = other.criteria
+  const length = objCriteria.length
+  const ordersLength = orders.length
+
+  while (++index < length) {
```

Comment:
As originally written, it can short-circuit if length is 0, so that none of the work in the while block is executed. Changing this to a do-while loop will not work because do-while guarantees at least 1 execution of the block.